### ISSUE TITLE
Re-used images only show up in editor after reload

### ISSUE THREAD

**Issue by snooptl** [codevise/pageflow#1209]
When inserting an image into a story which was already in use in another story (via "Reuse"), the list of available image files in this story is not immediately updated. Only after a manual browser-reload the image is available in the list. So the assignment works, but the visible list is not automatically updated.

I already checked the error logs and the JS-console in the browser, without finding any corresponding entries. We can reproduce this issue in all browsers tested (Chrome 76, Firefox 69 and Safari 12.1). I am unsure whether this is a general Bug in Pageflow or if it is specific to our self-hosted instance (v14.4.0).

I would appreciate any hints or useful insight. Please let me know if further information is required to narrow down the exact nature of the problem.

**Comment by tf** [comment]
Sorry for the late reply. I cannot reproduce the issue on our instances. Do you see any errors in the browser console after selecting an image for reuse? Reusing a file should trigger a reload of the generated entry stylesheet which contains the background-image CSS rules. If you have the network panel open in the dev tools while selecting a file, you should see a request of the form:
`.../revisions/611.css?p=14.4.0&v=pageflow%2Frevisions%2F611&reload=1568734182204`.

If you have configured the asset host to use a CDN, you might need to check in the CDN configuration that the query string is part of the cache key. Otherwise you might see stale data. On the other hand, this should cause the problem to persist across reloads as well...

**Comment by snooptl** [comment]
Hey tf,

Thanks for the reply.

>Do you see any errors in the browser console after selecting an image for reuse?

Unfortunately not.
  

> If you have the network panel open in the dev tools while selecting a file, you should see a request of the form:
.../revisions/611.css?p=14.4.0&v=pageflow%2Frevisions%2F611&reload=1568734182204.

I only see this request on the initial pageload. Not while/after inserting an image via "reuse". The only thing i see during the insertion is an XHR-Request to editor/entries/118/files/image_files/reuse 
The response body is:
```
Turbolinks.clearCache()
Turbolinks.visit("http://DOMAIN.com/entries/118", {"action":"replace"})
```

The log on the server contains these entries:
```
I, [2019-09-19T10:17:15.461056 #23755]  INFO -- : Started POST "/editor/entries/118/files/image_files/reuse" for XXX.XXX.XXX.XXX at 2019-09-19 10:17:15 +0200
I, [2019-09-19T10:17:15.463124 #23755]  INFO -- : Processing by Pageflow::Editor::FilesController#reuse as JSON
I, [2019-09-19T10:17:15.463253 #23755]  INFO -- :   Parameters: {"file_reuse"=>{"other_entry_id"=>14, "file_id"=>72}, "entry_id"=>"118", "collection_name"=>"image_files", "file"=>{"file_reuse"=>{"other_entry_id"=>14, "file_id"=>72}}}
I, [2019-09-19T10:17:15.493504 #23755]  INFO -- : Redirected to http://DOMAIN.com/entries/118
```

Thanks for your help

**Comment by tf** [comment]
Interesting. Since our instances do not use Turbolinks, the `reuse` request is just a plain redirect in our case. I wonder if Turbolinks might somehow mess things up here.

The overall flow should look like this: The `success` handler of for the `reuse` XHR request [triggers a `use:files` event](https://github.com/codevise/pageflow/blob/23e84c71671c12cade071da983d61eecf1d4e6ef/app/assets/javascripts/pageflow/editor/models/entry.js#L100). That event [triggers the stylesheet reload](https://github.com/codevise/pageflow/blob/23e84c71671c12cade071da983d61eecf1d4e6ef/app/assets/javascripts/pageflow/editor/initializers/stylesheet_reloading.js#L9). 

If you do not see the stylesheet reload, then it is clear that images will not be displayed correctly. The remaining question is why the reload doesn't happen. Maybe you can use the JS debugger to trace down at which points the process fails.

**Comment by snooptl** [comment]
Hey tf,

I was somehow under the impression that turbolinks is a dependency added by pageflow. After inspecting our Gemfile I realised, that I was mistaken. **Removing turbolinks seems to fix the problem**.  Now I see the request for the stylesheet and the new image pops up in the image-select-window.

Thank you for your help!

**Comment by tf** [comment]
Happy to hear that! :+1:
